### PR TITLE
Jaas security scope

### PR DESCRIPTION
**Pull request by ale8k**
Provides a docco for security scope within JAAS.

### CODE REVIEW

File: explanation/jaas_security_scope.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+and prevent data interception or tampering.
+
+Additionally, when communicating with any controller, JAAS uses token based authorisation whilst
+additionally acting as the IdP (Identity Provider) for said tokens.
+
+Access control and authentication
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+JAAS provides an abstraction layer of access control atop of Juju. JAAS does this by backing its users
+with an IdP (Identity Provider), this particular IdP is called Candid which handles the authorisation 
+and authentication for JIMM (Juju Intelligent Model Manager, the backend to JAAS). Furthermore, JAAS is
```
Comment (kian99) [inline explanation/jaas_security_scope.rst:18]: Doesn't Candid just handle authentication? JIMM handles authorization.

File: explanation/jaas_security_scope.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Access control and authentication
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+JAAS provides an abstraction layer of access control atop of Juju. JAAS does this by backing its users
+with an IdP (Identity Provider), this particular IdP is called Candid which handles the authorisation 
+and authentication for JIMM (Juju Intelligent Model Manager, the backend to JAAS). Furthermore, JAAS is 
+capable of many authentication and authorisation types, ranging from users to agents.
+
+Auditing and logging
+~~~~~~~~~~~~~~~~~~~~
+JAAS provides audit logs of all access to each model managed by JAAS and by which user.
```
Comment (kian99) [inline explanation/jaas_security_scope.rst:23]: s/by which user/including information on which user performed the action
or something similar

**Review by kian99 — APPROVED** [review]
Looks good with some suggestions.

**Review by kian99 — APPROVED** [review]
Looks good with some suggestions.

File: explanation/jaas_security_scope.rst [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+JAAS ensures secure communication between controllers (management nodes) and models (namespaces)
+by means of authorisation, authentication and TLS encryption to prevent unauthorised access
+and prevent data interception or tampering.
+
+Additionally, when communicating with any controller, JAAS uses token based authorisation whilst
+additionally acting as the IdP (Identity Provider) for said tokens.
+
+Access control and authentication
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+JAAS provides an abstraction layer of access control atop of Juju. JAAS does this by backing its users
```
[the remark this answers is not in the document]
Comment (ale8k) [reply explanation/jaas_security_scope.rst:16]: Sure, will change

File: explanation/jaas_security_scope.rst (hunk 1 above)
Comment (ale8k) [reply explanation/jaas_security_scope.rst:18]: I think... candid will authenticate a user and provide a macaroon with some basic caveats which I guess are a form of authorisation? We don't have centralised authz but I'll reword it as candid = authn, jimm = authz as it's more of an implementation detail probably

File: explanation/jaas_security_scope.rst (hunk 2 above)
Comment (ale8k) [reply explanation/jaas_security_scope.rst:23]: Ah ok sounds better, done

**Pull request merged**